auth/src/main/java/com/firebase/ui/auth/provider/FacebookProvider.java

```diff
@@ -1,212 +0,0 @@
-import com.firebase.ui.auth.data.model.User;
-import com.firebase.ui.auth.util.ExtraConstants;
-import com.google.firebase.auth.AuthCredential;
-import com.google.firebase.auth.FacebookAuthProvider;
-
-import org.json.JSONException;
-import org.json.JSONObject;
-
-import java.util.ArrayList;
-import java.util.List;
-
-public class FacebookProvider implements IdpProvider, FacebookCallback<LoginResult> {
-    private static final String TAG = "FacebookProvider";
-    private static final String EMAIL = "email";
-    private static final String PUBLIC_PROFILE = "public_profile";
-
-    private static CallbackManager sCallbackManager;
-
-    private final List<String> mScopes;
-    // DO NOT USE DIRECTLY: see onSuccess(String, LoginResult) and onFailure(Bundle) below
-    private IdpCallback mCallbackObject;
-
-    public FacebookProvider(AuthUI.IdpConfig idpConfig, @StyleRes int theme) {
-        List<String> scopes = idpConfig.getParams()
-                .getStringArrayList(ExtraConstants.EXTRA_FACEBOOK_PERMISSIONS);
-        if (scopes == null) {
-            mScopes = new ArrayList<>();
-        } else {
-            mScopes = scopes;
-        }
-        WebDialog.setWebDialogTheme(theme);
-    }
-
-    public static AuthCredential createAuthCredential(IdpResponse response) {
-        if (!response.getProviderType().equals(FacebookAuthProvider.PROVIDER_ID)) {
-            return null;
-        }
-        return FacebookAuthProvider.getCredential(response.getIdpToken());
-    }
-
-    @Override
-    public String getName(Context context) {
-        return context.getString(R.string.fui_idp_name_facebook);
-    }
-
-    @Override
-    @LayoutRes
-    public int getButtonLayout() {
-        return R.layout.fui_idp_button_facebook;
-    }
-
-    @Override
-    public void startLogin(Activity activity) {
-        sCallbackManager = CallbackManager.Factory.create();
-        LoginManager loginManager = LoginManager.getInstance();
-        loginManager.registerCallback(sCallbackManager, this);
-
-        List<String> permissionsList = new ArrayList<>(mScopes);
-
-        // Ensure we have email and public_profile scopes
-        if (!permissionsList.contains(EMAIL)) {
-            permissionsList.add(EMAIL);
-        }
-
-        if (!permissionsList.contains(PUBLIC_PROFILE)) {
-            permissionsList.add(PUBLIC_PROFILE);
-        }
-
-        // Log in with permissions
-        loginManager.logInWithReadPermissions(activity, permissionsList);
-    }
-
-    @Override
-    public void setAuthenticationCallback(IdpCallback callback) {
-        mCallbackObject = callback;
-    }
-
-    @Override
-    public void onActivityResult(int requestCode, int resultCode, Intent data) {
-        if (sCallbackManager != null) {
-            sCallbackManager.onActivityResult(requestCode, resultCode, data);
-        }
-    }
-
-    @Override
-    public void onSuccess(final LoginResult loginResult) {
-        GraphRequest request = GraphRequest.newMeRequest(
```

Comment:
Could you do me a favor and mark out anywhere in the new `Handler` classes that you've made a major logic change from the `Provider`?  For instance I notice this whole block is pretty much exactly the same in the new class but as a reviewer I still felt like I had to read it since Github preserved no diff.